packages/evolution-backend/src/api/helpers/downloadCsvFile.ts

```diff
@@ -52,16 +60,24 @@ export const downloadCsvFile = async ({
                     reject(error);
                 })
                 .on('data', (row) => {
-                    // Prepare the CSV row with default values
-                    const csvRow: { [key: string]: string | number | boolean | null } = {
-                        interview_id: row.interview_id,
-                        user_email: row.user_email,
-                        user_username: row.user_username
-                    };
+                    const csvRow: { [key: string]: string | number | boolean | null } = {};
+
+                    // Add if not restricted
+                    if (!restrictedResponses?.includes('interview_id')) {
+                        csvRow.interview_id = row.interview_id;
+                    }
+                    if (!restrictedResponses?.includes('user_email')) {
+                        csvRow.user_email = row.user_email;
+                    }
+                    if (!restrictedResponses?.includes('user_username')) {
+                        csvRow.user_username = row.user_username;
+                    }
 
+                    // Add requested responses except restricted responses
                     requestedResponses.forEach((response) => {
-                        // We use lodash's get function to safely access nested properties
-                        csvRow[response] = get(row.responses, response, null);
+                        if (!restrictedResponses?.includes(response as RestrictedResponse)) {
```

Comment:
technically, given the type of the RestrictedResponse, there should be no other fields to exclude. But you could type `restrictedResponses` as `string[]` so that you can put anything there (though if you exclude them from the file, why include them in the requestedResponses in the first place... maybe this if doesn't need to be there)

Reply:
Yeah, not sure. Why type it string[] if you can have a nice enum?

Reply:
@tahini 
Je ne suis pas sûr de comprendre 

> then you don't need this if here

 La fonction génère pas default, quelques fields qui je crois vont être utiliser dans la plupart des fichiers à télécharger.

Par contre, parfois (pour une question de confidentialité), on veut exclure certains fields (pour partager à un partenaire par exemple). 

En d'autres mots, le requestedResponses (peut être qu'il faut renommer requestedFields): quels champs tu veux télécharger.
Le excludedFields: quel champs parmis ceux déjà par default, on veut exclure.

Le requestedResponses et le excludedFields ne devrait pas être le même field.

Reply:
> Le requestedResponses et le excludedFields ne devrait pas être le même field.

Exactement! Donc on ne devrait pas avoir de champs commun entre ces requestedResponses et excludedFields. Ce if-là va enlever les champs qui sont requested ET excluded. Est-ce utile?

Reply:
Corrigé.